### PR TITLE
Add stats to HealthCheck, and export its cache.

### PR DESCRIPTION
**Pull request by guoliang100**
@alainjobart @sougou @guokeno0

### CODE REVIEW

File: go/vt/discovery/healthcheck.go [hunk 1]
```diff
@@ -334,11 +355,89 @@ func (hc *HealthCheckImpl) deleteEndPointFromTargetProtected(target *pbq.Target,
...
 }
 
+// EndPointsCacheStatus is the current endpoints for a cell/target.
+type EndPointsCacheStatus struct {
+	Cell           string
+	Target         *pbq.Target
+	EndPointsStats []*EndPointStats
+}
+
+// StatusAsHTML returns an HTML version of the status.
```
Comment (guokeno0) [inline go/vt/discovery/healthcheck.go]: How about add a TODO to change this status to reflect the End-to-End information about the endpoints. E.g.

bns_of_endpoint1 querys/errors

**Comment by guokeno0** [comment]
LGTM on the varz part

File: go/vt/discovery/healthcheck.go (hunk 1 above)
Comment (guoliang100) [inline go/vt/discovery/healthcheck.go]: Done

**Comment by alainjobart** [comment]
LGTM. Let's see with usage if that's the metrics we want to track.

**Pull request merged**